docs/demos/kubeadm-bootstrap/kubeadm-bootstrap.sh

```diff
@@ -0,0 +1,108 @@
+OWD="${PWD}"
+WORK_DIR="${OWD}/demo-kubeadm-bootstrap"
+KUBERNETES_DIR="${WORK_DIR}/etc_kubernetes"
+CERTIFICATES_DIR="${KUBERNETES_DIR}/pki"
+VCERT_ZONE="TLS/SSL\Certificates\Kubernetes"
+VCERT_CA="${SCRIPT_DIR}/ca.venafi.crt"
+
+: ${VCERT_INI:?}
+: ${KIND:?}
+: ${KUBEADM:?}
+
+function log() {
+    echo >&2
+    echo "# $(date --rfc-3339=ns) :: ${*}" >&2
+}
+
+function kubeadm_init_phase_certs() {
+    while read cert; do
+        log "Generating CSR ${cert}"
+        local args=""
+        if [[ "${cert}" == "apiserver" ]]; then
+            args="--apiserver-cert-extra-sans=kind-control-plane,127.0.0.1"
+        fi
+        ${KUBEADM} init phase certs "${cert}" ${args} --csr-only --csr-dir "${CERTIFICATES_DIR}" --cert-dir "${CERTIFICATES_DIR}" >/dev/null
+        echo "${cert} ${CERTIFICATES_DIR}/${cert/#etcd-/etcd\/}.csr"
+    done
+}
+
+function vcert_enroll() {
+    while read nickname csr; do
+        cert="${csr/%.csr/.crt}"
+        policy="$(echo "${nickname}" | grep -o '\(client\|server\|peer\)$')"
+        # Special case for etcd-server which needs both server and client usage
+        # See https://clusterise.com/articles/kbp-2-certificates/ and
+        # https://kubernetes.io/docs/setup/best-practices/certificates/#all-certificates
+        if [[ "${nickname}" == "etcd-server" ]]; then
+            policy=peer
+        fi
+        log "Enrolling CSR ${csr} with nickname ${nickname} to ${cert}"
+        vcert enroll -z "${VCERT_ZONE}\\${policy}"  --config ${VCERT_INI} --nickname "${nickname}" --csr "file:${csr}" --cert-file "${cert}" >/dev/null
+    done
+}
+
+if [[ -d "${WORK_DIR}" ]]; then
+    log "Deleting previous work dir: ${WORK_DIR}"
+    rm -rf "${WORK_DIR}"
+fi
+
+log "Working in ${WORK_DIR}"
+mkdir -p "${WORK_DIR}"
+pushd "${WORK_DIR}"
+
+log "Creating certificates directory ${CERTIFICATES_DIR}"
+mkdir -p "${CERTIFICATES_DIR}"
+
+log "Generating certificate signing requests"
+kubeadm_init_phase_certs <<EOF | vcert_enroll
+  apiserver
+  apiserver-etcd-client
+  apiserver-kubelet-client
+  etcd-healthcheck-client
+  etcd-peer
+  etcd-server
+  front-proxy-client
+EOF
+
+log "Creating self-signed certificate authority"
+${KUBEADM} init phase certs ca --cert-dir "${PWD}/pki.self-signed" >/dev/null
+
+log "Creating Kubeconfigs"
+${KUBEADM} init phase kubeconfig all \
+           --cert-dir "${PWD}/pki.self-signed" \
+           --kubeconfig-dir "${KUBERNETES_DIR}" \
+           --control-plane-endpoint kind-control-plane \
+           --node-name kind-control-plane
+
+log "Setting Venafi CA in all kubeconfigs"
+venafi_ca_data=$(base64 -w 0 < "${SCRIPT_DIR}/ca.venafi.crt")
+find "${KUBERNETES_DIR}" -name '*.conf'  | \
+    xargs -n 1 -I {} -- \
+          kubectl --kubeconfig={} config set clusters.kubernetes.certificate-authority-data "${venafi_ca_data}"
+
+log "Creating SA"
+${KUBEADM} init phase certs sa --cert-dir "${CERTIFICATES_DIR}" >/dev/null
+
+log "Installing Venafi CA cert"
```

Comment:
Add a comment explaining that this must come after the creation of csrs, due to https://github.com/kubernetes/kubeadm/issues/2163